joint_trajectory_controller/test/test_trajectory_actions.cpp

```diff
@@ -417,50 +410,51 @@ TEST_P(TestTrajectoryActionsTestParameterized, test_state_tolerances_fail)
     rclcpp::Parameter("constraints.joint2.trajectory", state_tol),
     rclcpp::Parameter("constraints.joint3.trajectory", state_tol)};
 
-  SetUpExecutor(params);
+  // separate command from states -> immediate state tolerance fail
+  bool separate_cmd_and_state_values = true;
+  SetUpExecutor(params, separate_cmd_and_state_values);
   SetUpControllerHardware();
 
   std::shared_future<typename GoalHandle::SharedPtr> gh_future;
   // send goal
+  std::vector<std::vector<double>> points_positions{{{4.0, 5.0, 6.0}}, {{7.0, 8.0, 9.0}}};
   {
     std::vector<JointTrajectoryPoint> points;
     JointTrajectoryPoint point1;
     point1.time_from_start = rclcpp::Duration::from_seconds(0.0);
     point1.positions.resize(joint_names_.size());
 
-    point1.positions[0] = 4.0;
-    point1.positions[1] = 5.0;
-    point1.positions[2] = 6.0;
+    point1.positions = points_positions.at(0);
     points.push_back(point1);
 
     JointTrajectoryPoint point2;
     point2.time_from_start = rclcpp::Duration::from_seconds(0.1);
     point2.positions.resize(joint_names_.size());
 
-    point2.positions[0] = 7.0;
-    point2.positions[1] = 8.0;
-    point2.positions[2] = 9.0;
+    point2.positions = points_positions.at(1);
     points.push_back(point2);
 
     gh_future = sendActionGoal(points, 1.0, goal_options_);
   }
   controller_hw_thread_.join();
 
   EXPECT_TRUE(gh_future.get());
+  // TODO(christophfroehlich) sometimes the trajectory is overwritten by
```

Comment:
How can this happen here? `Holding position due to state tolerance violation` is in the if branch if there is no active goal.
```
4: [ RUN      ] PositionTrajectoryControllersActions/TestTrajectoryActionsTestParameterized.test_state_tolerances_fail/2
4: [WARN] [1689668022.296384234] [test_joint_trajectory_controller]: [Deprecated]: "allow_nonzero_velocity_at_trajectory_end" is set to true. The default behavior will change to false.
4: [INFO] [1689668022.297142446] [test_joint_trajectory_controller]: No specific joint names are used for command interfaces. Using 'joints' parameter.
4: [INFO] [1689668022.297155740] [test_joint_trajectory_controller]: Command interfaces are [position] and state interfaces are [position velocity acceleration].
4: [INFO] [1689668022.297162969] [test_joint_trajectory_controller]: Using 'splines' interpolation method.
4: [INFO] [1689668022.297494705] [test_joint_trajectory_controller]: Action status changes will be monitored at 20.00 Hz.
4: [INFO] [1689668022.599148369] [test_joint_trajectory_controller]: Received new action goal
4: [INFO] [1689668022.599193964] [test_joint_trajectory_controller]: Accepted new action goal
4: [ERROR] [1689668022.599247504] [test_joint_trajectory_controller]: Holding position due to state tolerance violation
4: [INFO] [1689668022.599315419] [test_joint_trajectory_controller]: Goal reached, success!
4: /workspaces/ros2_rolling_ws/src/ros2_controllers/joint_trajectory_controller/test/test_trajectory_actions.cpp:446: Failure
4: Expected equality of these values:
4:   rclcpp_action::ResultCode::ABORTED
4:     Which is: 1-byte object <06>
4:   common_resultcode_
4:     Which is: 1-byte object <04>
4: /workspaces/ros2_rolling_ws/src/ros2_controllers/joint_trajectory_controller/test/test_trajectory_actions.cpp:447: Failure
4: Expected equality of these values:
4:   control_msgs::action::FollowJointTrajectory_Result::PATH_TOLERANCE_VIOLATED
4:     Which is: -4
4:   common_action_result_code_
4:     Which is: 0
```

Reply:
Fixed with `rt_has_pending_goal_` boolean